#!/usr/bin/env python
PACKAGE = "transbot_linefollow"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("scale", int_t, 0, "scale", 1000, 0, 1000)
gen.add("Kp", int_t, 0, "Kp in PID", 40, 0, 100)
gen.add("Ki", int_t, 0, "Ki in PID", 0, 0, 100)
gen.add("Kd", int_t, 0, "Kd in PID", 12, 0, 100)
gen.add("Hmin", int_t, 0, "Hmin in HSV", 0, 0, 255)
gen.add("Smin", int_t, 0, "Smin in HSV", 85, 0, 255)
gen.add("Vmin", int_t, 0, "Vmin in HSV", 126, 0, 255)
gen.add("Hmax", int_t, 0, "Hmax in HSV", 9, 0, 255)
gen.add("Smax", int_t, 0, "Smax in HSV", 253, 0, 255)
gen.add("Vmax", int_t, 0, "Vmax in HSV", 255, 0, 255)
gen.add("linear", double_t, 0, "linear", 0.3, 0, 0.5)
gen.add("LaserAngle", int_t, 0, "LaserAngle", 30, 10, 180)
gen.add("ResponseDist", double_t, 0, "ResponseDist", 0.55, 0, 8)
exit(gen.generate(PACKAGE, "LineDetect", "LineDetectPID"))
